core/src/main/scala/io/delta/sql/parser/DeltaSqlParser.scala

```diff
@@ -201,6 +197,15 @@ class DeltaSqlAstBuilder extends DeltaSqlBaseBaseVisitor[AnyRef] {
       Option(ctx.table).map(visitTableIdentifier))
   }
 
+  override def visitShowCreateTable(ctx: DeltaSqlBaseParser.ShowCreateTableContext): LogicalPlan = withOrigin(ctx) {
+    val tableIdentifier = visitTableIdentifier(ctx.table)
+    if (DeltaLog.forTable(SparkSession.active, tableIdentifier).tableExists) {
```

Comment:
This is not the right approach to check whether it's a delta table. Could you check the code for `show columns`?